library/src/armory/metric.py

```diff
@@ -1,20 +1,30 @@
 from abc import ABC, abstractmethod
-from typing import TYPE_CHECKING, Optional
+from typing import TYPE_CHECKING, Any, Dict, Iterable, Optional
 
+from jsonpath_ng import parse
```

Comment:
style nitpick: `import jsonpath_ng; … {path: jsonpath_ng(path) for path in record_as_metrics}` since the symbol is only used once